src/django/api/views/v1/moderation_events.py

```diff
@@ -86,69 +103,81 @@ def retrieve(self, _,  pk=None):
         events = response.get("data", [])
 
         if len(events) == 0:
-            return Response(
-                data={
-                    "detail": 'The moderation event with the '
-                    'given uuid was not found.',
-                },
-                status=status.HTTP_404_NOT_FOUND,
+            raise NotFound(
+                detail=APIV1CommonErrorMessages.MODERATION_EVENT_NOT_FOUND
             )
 
         return Response(events[0])
 
     @handle_errors_decorator
     def partial_update(self, request, pk=None):
-        ModerationEventsService.validate_user_permissions(request)
-
         ModerationEventsService.validate_uuid(pk)
 
-        event = ModerationEventsService.fetch_moderation_event_by_uuid(
-            pk
-        )
+        event = ModerationEventsService.fetch_moderation_event_by_uuid(pk)
 
         serializer = ModerationEventUpdateSerializer(
-            event,
-            data=request.data,
-            partial=True
+            event, data=request.data, partial=True
         )
 
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data, status=status.HTTP_200_OK)
+        if not serializer.is_valid():
+            raise ValidationError(
+                {
+                    "detail": APIV1CommonErrorMessages.COMMON_REQ_BODY_ERROR,
+                    "errors": [serializer.errors],
+                }
+            )
+
+        serializer.save()
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    @handle_errors_decorator
```

Comment:
Since this decorator is primarily focused on OpenSearch-related errors or errors related to searching via OpenSearch, I think it makes sense either not to set it that or to create a new handler and rename this one to clearly reflect its purpose. One of the errors mentions input, which in the case of PATCH requests may be unusual to receive.

Reply:
Removed `handle_errors_decorator` from this action.